### PR TITLE
[pkg/pdatatest] add more test cases for trace

### LINKED ISSUE #18030: [pkg/pdatatest] add more test for traces
### Component(s)

pkg/pdatatest

### Is your feature request related to a problem? Please describe.

add more test for traces 

### Describe the solution you'd like

add more test for traces

### Describe alternatives you've considered

_No response_

### Additional context

partly of #17544

### PR DESCRIPTION
**Pull request by fatsheep9146**
Signed-off-by: Ziqi Zhao <zhaoziqi9146@gmail.com>

**Description:** <Describe what has changed.>
fix #18030
partly of #17544

### CODE REVIEW

**Review by bogdandrutu — APPROVED** [review]

**Comment by dmitryax** [comment]
@fatsheep9146 please rebase

**Comment by fatsheep9146** [comment]
> @fatsheep9146 please rebase

Done

File: pkg/pdatatest/ptracetest/traces.go [hunk 1]
```diff
@@ -287,8 +287,7 @@ func CompareSpan(expected, actual ptrace.Span) error {
 	}
 
 	if !reflect.DeepEqual(expected.Events(), actual.Events()) {
-		errs = multierr.Append(errs, fmt.Errorf("events doesn't match expected: %v, actual: %v",
-			expected.Events(), actual.Events()))
+		errs = multierr.Append(errs, fmt.Errorf("events doesn't match"))
```
Comment (dmitryax) [inline pkg/pdatatest/ptracetest/traces.go:290]: We need to expand this comparison. I submitted another issue. Feel free to take it if you're interested https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18049

File: pkg/pdatatest/ptracetest/traces_test.go [hunk 2]
```diff
@@ -75,7 +203,7 @@ func TestCompareTraces(t *testing.T) {
 			if tc.withoutOptions == nil {
 				assert.NoError(t, err)
 			} else {
-				assert.EqualError(t, tc.withoutOptions, err.Error())
+				assert.EqualError(t, err, tc.withoutOptions.Error())
```
Comment (dmitryax) [inline pkg/pdatatest/ptracetest/traces_test.go:206]: Thank's for fixing it. I thought the first argument was expected value, appeared to be it's the second one. Feel free to fix other tests if you have a chance

**Review by dmitryax — APPROVED** [review]
LGTM. Thanks @fatsheep9146

File: pkg/pdatatest/ptracetest/traces_test.go (hunk 2 above)
Comment (dmitryax) [reply pkg/pdatatest/ptracetest/traces_test.go:206]: Actually, nevermind. I'll fix in a PR that I'm currently working on

File: pkg/pdatatest/ptracetest/traces.go (hunk 1 above)
Comment (fatsheep9146) [reply pkg/pdatatest/ptracetest/traces.go:290]: ok, I take this @dmitryax

File: .chloggen/add_test_for_traces.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Adds more test cases for traces 
+
+# One or more tracking issues related to the change
+issues: [18030]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
```
Comment (dmitryax) [inline .chloggen/add_test_for_traces.yaml:16]: Forgot to mention that we shouldn't add changelog entries for test coverage since it's not a user-facing change. Please submit another PR to remove this entry
